Prepare automated tests for Camel LSP for VS Code

djelinek opened the pull request:
@apupier Hi, please have a look at least on package.json 'scripts' part, I did some modifications there. Thanks

djelinek commented:
Related to #137

apupier commented on package.json:
```diff
@@ -111,14 +111,17 @@
+    "vscode:prepublish": "npm run compile",
+    "compile": "tsc -p ./",
+    "watch": "tsc -watch -p ./",
+    "preinstall": "npm i --save mvn-artifact-download && node ./scripts/preinstall.js",
```
isn't to avoid having to call "npm install --ignore-scripts"?

djelinek replied:
I suppose yes, I added this because if we wanted to execute npm install it will always failed on missing mvn dependency and I did not found how to solved that other way.

apupier replied:
it sounds cool.
May I ask you to create a different PR just for that to separate added functionalities please?
it will ease review to have them separately.
Also please update:
- the travis.yml file and remove the` npm install --ignore-scripts`
- the contributing.md and remove the  npm install --ignore-scripts`

djelinek replied:
sure, I will do that :slightly_smiling_face:

djelinek replied:
Created #140

apupier commented on package.json:
```diff
@@ -111,14 +111,17 @@
     "test": "node ./node_modules/vscode/bin/test",
-    "tslint": "gulp tslint"
+    "tslint": "gulp tslint",
+    "ui-test": "npm run compile && extest setup-and-run --code_version 1.38.1 out/ui-test/*.js -o src/ui-test/settings.json"
```
I'm wondering how/when we will update this version

djelinek replied:
for now it would be manual decision after local verification that UI tests are working

apupier commented on package.json:
```diff
@@ -111,14 +111,17 @@
     "test": "node ./node_modules/vscode/bin/test",
-    "tslint": "gulp tslint"
+    "tslint": "gulp tslint",
+    "ui-test": "npm run compile && extest setup-and-run --code_version 1.38.1 out/ui-test/*.js -o src/ui-test/settings.json"
```
Should we let it using the latest version?
or if we specify a version maybe we need to have itr as a parameter so that we can test severla versions directly on Ci? (or it might be overengineered and better to handle it in another iteration)

I'm thinking loud :-)

djelinek replied:
yeah, very good question. I think zou are right I can remove '--code_version' parameter for now and let it uses latest version (which is currently same 1.38.1). For next iteration I like your idea to have it parametrized for testing lets say latest two or three vscode versions back. We will also see how much will allow us new version of vscode-extension-tester.

apupier replied:
ok, I vote for remove the --code_version for now. As VS Code is upgrading automatically by default it will allow us to catch issues relatively fast.
I let you create an issue for another iteration with several version to be tested.
I see it well as different travis stage build parameterized with:
- the minimally declared version
- the latest stable version
- the "insiders" build

djelinek replied:
removed '--code_version' -> using latest version of vscode

apupier commented:
tests are failing locally (will investigate a bit but if you have an idea...)

```
Detected user defined code settings
Writing code settings to /home/apupier/git/camel-lsp-client-vscode/test-resources/settings/User/settings.json
  Camel LSP - XML DSL Support
    Extension Activation
      ✓ Language Support for Apache Camel is installed and started (229ms)
    Code Completion
      ✓ Open "camel.xml" file inside Editor View (178ms)
      1) Code completion is working for component schemes (the part before the ":")
      2) Code completion is working for endpoint options (the part after the "?")
      3) Code completion is working for additional endpoint options (the part after "&")
    Endpoint options filtering
      4) "before all" hook
    Diagnostics for Camel URIs
      5) "before all" hook
    Auto-completion for referenced components IDs
      6) "before all" hook

Waiting for all webdriver tasks to finish
  7) "after all" hook

  2 passing (31s)
  7 failing

  1) Camel LSP - XML DSL Support
       Code Completion
         Code completion is working for component schemes (the part before the ":"):
     Error: Line number 3 does not exist
      at TextEditor.<anonymous> (node_modules/vscode-extension-tester/out/webdriver/components/editor/TextEditor.js:161:23)
      at Generator.next (<anonymous>)
      at fulfilled (node_modules/vscode-extension-tester/out/webdriver/components/editor/TextEditor.js:4:58)
      at process._tickCallback (internal/process/next_tick.js:68:7)

  2) Camel LSP - XML DSL Support
       Code Completion
         Code completion is working for endpoint options (the part after the "?"):
     Uncaught NoSuchWindowError: no such window: target window already closed
from unknown error: web view not found
  (Session info: chrome=69.0.3497.128)
  (Driver info: chromedriver=2.38.552522 (437e6fbedfa8762dec75e2c5b3ddb86763dc9dcb),platform=Linux 5.2.13-200.fc30.x86_64 x86_64)
      at Object.checkLegacyResponse (node_modules/selenium-webdriver/lib/error.js:546:15)
      at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:509:13)
      at doSend.then.response (node_modules/selenium-webdriver/lib/http.js:441:30)
      at process._tickCallback (internal/process/next_tick.js:68:7)
  From: Task: WebDriver.findElement(By(css selector, body))
      at Driver.schedule (node_modules/selenium-webdriver/lib/webdriver.js:807:17)
      at Driver.findElement (node_modules/selenium-webdriver/lib/webdriver.js:1014:17)
      at new AbstractElement (node_modules/vscode-extension-tester/out/webdriver/components/AbstractElement.js:27:43)
      at new EditorView (node_modules/vscode-extension-tester/out/webdriver/components/editor/EditorView.js:21:9)
      at Context.<anonymous> (out/ui-test/xml_completion_support.js:73:73)
      at Generator.next (<anonymous>)
      at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:8:71
      at new Promise (<anonymous>)
      at __awaiter (out/ui-test/xml_completion_support.js:4:12)
      at Context.<anonymous> (out/ui-test/xml_completion_support.js:71:20)

  3) Camel LSP - XML DSL Support
       Code Completion
         Code completion is working for additional endpoint options (the part after "&"):
     Uncaught NoSuchWindowError: no such window: target window already closed
from unknown error: web view not found
  (Session info: chrome=69.0.3497.128)
  (Driver info: chromedriver=2.38.552522 (437e6fbedfa8762dec75e2c5b3ddb86763dc9dcb),platform=Linux 5.2.13-200.fc30.x86_64 x86_64)
      at Object.checkLegacyResponse (node_modules/selenium-webdriver/lib/error.js:546:15)
      at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:509:13)
      at doSend.then.response (node_modules/selenium-webdriver/lib/http.js:441:30)
      at process._tickCallback (internal/process/next_tick.js:68:7)
  From: Task: WebDriver.findElement(By(css selector, body))
      at Driver.schedule (node_modules/selenium-webdriver/lib/webdriver.js:807:17)
      at Driver.findElement (node_modules/selenium-webdriver/lib/webdriver.js:1014:17)
      at new AbstractElement (node_modules/vscode-extension-tester/out/webdriver/components/AbstractElement.js:27:43)
      at new EditorView (node_modules/vscode-extension-tester/out/webdriver/components/editor/EditorView.js:21:9)
      at Context.<anonymous> (out/ui-test/xml_completion_support.js:85:73)
      at Generator.next (<anonymous>)
      at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:8:71
      at new Promise (<anonymous>)
      at __awaiter (out/ui-test/xml_completion_support.js:4:12)
      at Context.<anonymous> (out/ui-test/xml_completion_support.js:83:20)

  4) Camel LSP - XML DSL Support
       Endpoint options filtering
         "before all" hook:
     Uncaught NoSuchWindowError: no such window: target window already closed
from unknown error: web view not found
  (Session info: chrome=69.0.3497.128)
  (Driver info: chromedriver=2.38.552522 (437e6fbedfa8762dec75e2c5b3ddb86763dc9dcb),platform=Linux 5.2.13-200.fc30.x86_64 x86_64)
      at Object.checkLegacyResponse (node_modules/selenium-webdriver/lib/error.js:546:15)
      at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:509:13)
      at doSend.then.response (node_modules/selenium-webdriver/lib/http.js:441:30)
      at process._tickCallback (internal/process/next_tick.js:68:7)
  From: Task: WebDriver.findElement(By(css selector, body))
      at Driver.schedule (node_modules/selenium-webdriver/lib/webdriver.js:807:17)
      at Driver.findElement (node_modules/selenium-webdriver/lib/webdriver.js:1014:17)
      at new AbstractElement (node_modules/vscode-extension-tester/out/webdriver/components/AbstractElement.js:27:43)
      at new EditorView (node_modules/vscode-extension-tester/out/webdriver/components/editor/EditorView.js:21:9)
      at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:184:19
      at Generator.next (<anonymous>)
      at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:8:71
      at new Promise (<anonymous>)
      at __awaiter (out/ui-test/xml_completion_support.js:4:12)
      at cleanEnvironment (out/ui-test/xml_completion_support.js:183:16)

  5) Camel LSP - XML DSL Support
       Diagnostics for Camel URIs
         "before all" hook:
     Uncaught NoSuchWindowError: no such window: target window already closed
from unknown error: web view not found
  (Session info: chrome=69.0.3497.128)
  (Driver info: chromedriver=2.38.552522 (437e6fbedfa8762dec75e2c5b3ddb86763dc9dcb),platform=Linux 5.2.13-200.fc30.x86_64 x86_64)
      at Object.checkLegacyResponse (node_modules/selenium-webdriver/lib/error.js:546:15)
      at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:509:13)
      at doSend.then.response (node_modules/selenium-webdriver/lib/http.js:441:30)
      at process._tickCallback (internal/process/next_tick.js:68:7)
  From: Task: WebDriver.findElement(By(css selector, body))
      at Driver.schedule (node_modules/selenium-webdriver/lib/webdriver.js:807:17)
      at Driver.findElement (node_modules/selenium-webdriver/lib/webdriver.js:1014:17)
      at new AbstractElement (node_modules/vscode-extension-tester/out/webdriver/components/AbstractElement.js:27:43)
      at new EditorView (node_modules/vscode-extension-tester/out/webdriver/components/editor/EditorView.js:21:9)
      at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:184:19
      at Generator.next (<anonymous>)
      at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:8:71
      at new Promise (<anonymous>)
      at __awaiter (out/ui-test/xml_completion_support.js:4:12)
      at cleanEnvironment (out/ui-test/xml_completion_support.js:183:16)

  6) Camel LSP - XML DSL Support
       Auto-completion for referenced components IDs
         "before all" hook:
     Uncaught NoSuchWindowError: no such window: target window already closed
from unknown error: web view not found
  (Session info: chrome=69.0.3497.128)
  (Driver info: chromedriver=2.38.552522 (437e6fbedfa8762dec75e2c5b3ddb86763dc9dcb),platform=Linux 5.2.13-200.fc30.x86_64 x86_64)
      at Object.checkLegacyResponse (node_modules/selenium-webdriver/lib/error.js:546:15)
      at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:509:13)
      at doSend.then.response (node_modules/selenium-webdriver/lib/http.js:441:30)
      at process._tickCallback (internal/process/next_tick.js:68:7)
  From: Task: WebDriver.findElement(By(css selector, body))
      at Driver.schedule (node_modules/selenium-webdriver/lib/webdriver.js:807:17)
      at Driver.findElement (node_modules/selenium-webdriver/lib/webdriver.js:1014:17)
      at new AbstractElement (node_modules/vscode-extension-tester/out/webdriver/components/AbstractElement.js:27:43)
      at new EditorView (node_modules/vscode-extension-tester/out/webdriver/components/editor/EditorView.js:21:9)
      at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:184:19
      at Generator.next (<anonymous>)
      at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:8:71
      at new Promise (<anonymous>)
      at __awaiter (out/ui-test/xml_completion_support.js:4:12)
      at cleanEnvironment (out/ui-test/xml_completion_support.js:183:16)

  7) "after all" hook:
     Uncaught NoSuchWindowError: no such window: target window already closed
from unknown error: web view not found
  (Session info: chrome=69.0.3497.128)
  (Driver info: chromedriver=2.38.552522 (437e6fbedfa8762dec75e2c5b3ddb86763dc9dcb),platform=Linux 5.2.13-200.fc30.x86_64 x86_64)
      at Object.checkLegacyResponse (node_modules/selenium-webdriver/lib/error.js:546:15)
      at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:509:13)
      at doSend.then.response (node_modules/selenium-webdriver/lib/http.js:441:30)
      at process._tickCallback (internal/process/next_tick.js:68:7)
  From: Task: WebDriver.findElement(By(css selector, body))
      at Driver.schedule (node_modules/selenium-webdriver/lib/webdriver.js:807:17)
      at Driver.findElement (node_modules/selenium-webdriver/lib/webdriver.js:1014:17)
      at new AbstractElement (node_modules/vscode-extension-tester/out/webdriver/components/AbstractElement.js:27:43)
      at new EditorView (node_modules/vscode-extension-tester/out/webdriver/components/editor/EditorView.js:21:9)
      at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:184:19
      at Generator.next (<anonymous>)
      at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:8:71
      at new Promise (<anonymous>)
      at __awaiter (out/ui-test/xml_completion_support.js:4:12)
      at cleanEnvironment (out/ui-test/xml_completion_support.js:183:16)
```

apupier commented:
when adding `console.log(await editor.getText());` in the test, only the first line is logged:

```
      ✓ Open "camel.xml" file inside Editor View (134ms)
<camelContext id="uitest-context" xmlns="http://camel.apache.org/schema/spring">

      1) Code completion is working for component schemes (the part before the ":")
```

when i look at the played test, it seems that the correct test is displayed.
Do we have screenshots available on test failure to check it?

//cc @jrichter1

djelinek commented:
@apupier I am afraid that answer is not, we don't have screenshots for test failures

djelinek commented:
> tests are failing locally (will investigate a bit but if you have an idea...)
> 
> ```
> Detected user defined code settings
> Writing code settings to /home/apupier/git/camel-lsp-client-vscode/test-resources/settings/User/settings.json
>   Camel LSP - XML DSL Support
>     Extension Activation
>       ✓ Language Support for Apache Camel is installed and started (229ms)
>     Code Completion
>       ✓ Open "camel.xml" file inside Editor View (178ms)
>       1) Code completion is working for component schemes (the part before the ":")
>       2) Code completion is working for endpoint options (the part after the "?")
>       3) Code completion is working for additional endpoint options (the part after "&")
>     Endpoint options filtering
>       4) "before all" hook
>     Diagnostics for Camel URIs
>       5) "before all" hook
>     Auto-completion for referenced components IDs
>       6) "before all" hook
> 
> Waiting for all webdriver tasks to finish
>   7) "after all" hook
> 
>   2 passing (31s)
>   7 failing
> 
>   1) Camel LSP - XML DSL Support
>        Code Completion
>          Code completion is working for component schemes (the part before the ":"):
>      Error: Line number 3 does not exist
>       at TextEditor.<anonymous> (node_modules/vscode-extension-tester/out/webdriver/components/editor/TextEditor.js:161:23)
>       at Generator.next (<anonymous>)
>       at fulfilled (node_modules/vscode-extension-tester/out/webdriver/components/editor/TextEditor.js:4:58)
>       at process._tickCallback (internal/process/next_tick.js:68:7)
> 
>   2) Camel LSP - XML DSL Support
>        Code Completion
>          Code completion is working for endpoint options (the part after the "?"):
>      Uncaught NoSuchWindowError: no such window: target window already closed
> from unknown error: web view not found
>   (Session info: chrome=69.0.3497.128)
>   (Driver info: chromedriver=2.38.552522 (437e6fbedfa8762dec75e2c5b3ddb86763dc9dcb),platform=Linux 5.2.13-200.fc30.x86_64 x86_64)
>       at Object.checkLegacyResponse (node_modules/selenium-webdriver/lib/error.js:546:15)
>       at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:509:13)
>       at doSend.then.response (node_modules/selenium-webdriver/lib/http.js:441:30)
>       at process._tickCallback (internal/process/next_tick.js:68:7)
>   From: Task: WebDriver.findElement(By(css selector, body))
>       at Driver.schedule (node_modules/selenium-webdriver/lib/webdriver.js:807:17)
>       at Driver.findElement (node_modules/selenium-webdriver/lib/webdriver.js:1014:17)
>       at new AbstractElement (node_modules/vscode-extension-tester/out/webdriver/components/AbstractElement.js:27:43)
>       at new EditorView (node_modules/vscode-extension-tester/out/webdriver/components/editor/EditorView.js:21:9)
>       at Context.<anonymous> (out/ui-test/xml_completion_support.js:73:73)
>       at Generator.next (<anonymous>)
>       at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:8:71
>       at new Promise (<anonymous>)
>       at __awaiter (out/ui-test/xml_completion_support.js:4:12)
>       at Context.<anonymous> (out/ui-test/xml_completion_support.js:71:20)
> 
>   3) Camel LSP - XML DSL Support
>        Code Completion
>          Code completion is working for additional endpoint options (the part after "&"):
>      Uncaught NoSuchWindowError: no such window: target window already closed
> from unknown error: web view not found
>   (Session info: chrome=69.0.3497.128)
>   (Driver info: chromedriver=2.38.552522 (437e6fbedfa8762dec75e2c5b3ddb86763dc9dcb),platform=Linux 5.2.13-200.fc30.x86_64 x86_64)
>       at Object.checkLegacyResponse (node_modules/selenium-webdriver/lib/error.js:546:15)
>       at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:509:13)
>       at doSend.then.response (node_modules/selenium-webdriver/lib/http.js:441:30)
>       at process._tickCallback (internal/process/next_tick.js:68:7)
>   From: Task: WebDriver.findElement(By(css selector, body))
>       at Driver.schedule (node_modules/selenium-webdriver/lib/webdriver.js:807:17)
>       at Driver.findElement (node_modules/selenium-webdriver/lib/webdriver.js:1014:17)
>       at new AbstractElement (node_modules/vscode-extension-tester/out/webdriver/components/AbstractElement.js:27:43)
>       at new EditorView (node_modules/vscode-extension-tester/out/webdriver/components/editor/EditorView.js:21:9)
>       at Context.<anonymous> (out/ui-test/xml_completion_support.js:85:73)
>       at Generator.next (<anonymous>)
>       at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:8:71
>       at new Promise (<anonymous>)
>       at __awaiter (out/ui-test/xml_completion_support.js:4:12)
>       at Context.<anonymous> (out/ui-test/xml_completion_support.js:83:20)
> 
>   4) Camel LSP - XML DSL Support
>        Endpoint options filtering
>          "before all" hook:
>      Uncaught NoSuchWindowError: no such window: target window already closed
> from unknown error: web view not found
>   (Session info: chrome=69.0.3497.128)
>   (Driver info: chromedriver=2.38.552522 (437e6fbedfa8762dec75e2c5b3ddb86763dc9dcb),platform=Linux 5.2.13-200.fc30.x86_64 x86_64)
>       at Object.checkLegacyResponse (node_modules/selenium-webdriver/lib/error.js:546:15)
>       at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:509:13)
>       at doSend.then.response (node_modules/selenium-webdriver/lib/http.js:441:30)
>       at process._tickCallback (internal/process/next_tick.js:68:7)
>   From: Task: WebDriver.findElement(By(css selector, body))
>       at Driver.schedule (node_modules/selenium-webdriver/lib/webdriver.js:807:17)
>       at Driver.findElement (node_modules/selenium-webdriver/lib/webdriver.js:1014:17)
>       at new AbstractElement (node_modules/vscode-extension-tester/out/webdriver/components/AbstractElement.js:27:43)
>       at new EditorView (node_modules/vscode-extension-tester/out/webdriver/components/editor/EditorView.js:21:9)
>       at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:184:19
>       at Generator.next (<anonymous>)
>       at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:8:71
>       at new Promise (<anonymous>)
>       at __awaiter (out/ui-test/xml_completion_support.js:4:12)
>       at cleanEnvironment (out/ui-test/xml_completion_support.js:183:16)
> 
>   5) Camel LSP - XML DSL Support
>        Diagnostics for Camel URIs
>          "before all" hook:
>      Uncaught NoSuchWindowError: no such window: target window already closed
> from unknown error: web view not found
>   (Session info: chrome=69.0.3497.128)
>   (Driver info: chromedriver=2.38.552522 (437e6fbedfa8762dec75e2c5b3ddb86763dc9dcb),platform=Linux 5.2.13-200.fc30.x86_64 x86_64)
>       at Object.checkLegacyResponse (node_modules/selenium-webdriver/lib/error.js:546:15)
>       at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:509:13)
>       at doSend.then.response (node_modules/selenium-webdriver/lib/http.js:441:30)
>       at process._tickCallback (internal/process/next_tick.js:68:7)
>   From: Task: WebDriver.findElement(By(css selector, body))
>       at Driver.schedule (node_modules/selenium-webdriver/lib/webdriver.js:807:17)
>       at Driver.findElement (node_modules/selenium-webdriver/lib/webdriver.js:1014:17)
>       at new AbstractElement (node_modules/vscode-extension-tester/out/webdriver/components/AbstractElement.js:27:43)
>       at new EditorView (node_modules/vscode-extension-tester/out/webdriver/components/editor/EditorView.js:21:9)
>       at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:184:19
>       at Generator.next (<anonymous>)
>       at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:8:71
>       at new Promise (<anonymous>)
>       at __awaiter (out/ui-test/xml_completion_support.js:4:12)
>       at cleanEnvironment (out/ui-test/xml_completion_support.js:183:16)
> 
>   6) Camel LSP - XML DSL Support
>        Auto-completion for referenced components IDs
>          "before all" hook:
>      Uncaught NoSuchWindowError: no such window: target window already closed
> from unknown error: web view not found
>   (Session info: chrome=69.0.3497.128)
>   (Driver info: chromedriver=2.38.552522 (437e6fbedfa8762dec75e2c5b3ddb86763dc9dcb),platform=Linux 5.2.13-200.fc30.x86_64 x86_64)
>       at Object.checkLegacyResponse (node_modules/selenium-webdriver/lib/error.js:546:15)
>       at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:509:13)
>       at doSend.then.response (node_modules/selenium-webdriver/lib/http.js:441:30)
>       at process._tickCallback (internal/process/next_tick.js:68:7)
>   From: Task: WebDriver.findElement(By(css selector, body))
>       at Driver.schedule (node_modules/selenium-webdriver/lib/webdriver.js:807:17)
>       at Driver.findElement (node_modules/selenium-webdriver/lib/webdriver.js:1014:17)
>       at new AbstractElement (node_modules/vscode-extension-tester/out/webdriver/components/AbstractElement.js:27:43)
>       at new EditorView (node_modules/vscode-extension-tester/out/webdriver/components/editor/EditorView.js:21:9)
>       at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:184:19
>       at Generator.next (<anonymous>)
>       at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:8:71
>       at new Promise (<anonymous>)
>       at __awaiter (out/ui-test/xml_completion_support.js:4:12)
>       at cleanEnvironment (out/ui-test/xml_completion_support.js:183:16)
> 
>   7) "after all" hook:
>      Uncaught NoSuchWindowError: no such window: target window already closed
> from unknown error: web view not found
>   (Session info: chrome=69.0.3497.128)
>   (Driver info: chromedriver=2.38.552522 (437e6fbedfa8762dec75e2c5b3ddb86763dc9dcb),platform=Linux 5.2.13-200.fc30.x86_64 x86_64)
>       at Object.checkLegacyResponse (node_modules/selenium-webdriver/lib/error.js:546:15)
>       at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:509:13)
>       at doSend.then.response (node_modules/selenium-webdriver/lib/http.js:441:30)
>       at process._tickCallback (internal/process/next_tick.js:68:7)
>   From: Task: WebDriver.findElement(By(css selector, body))
>       at Driver.schedule (node_modules/selenium-webdriver/lib/webdriver.js:807:17)
>       at Driver.findElement (node_modules/selenium-webdriver/lib/webdriver.js:1014:17)
>       at new AbstractElement (node_modules/vscode-extension-tester/out/webdriver/components/AbstractElement.js:27:43)
>       at new EditorView (node_modules/vscode-extension-tester/out/webdriver/components/editor/EditorView.js:21:9)
>       at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:184:19
>       at Generator.next (<anonymous>)
>       at /home/apupier/git/camel-lsp-client-vscode/out/ui-test/xml_completion_support.js:8:71
>       at new Promise (<anonymous>)
>       at __awaiter (out/ui-test/xml_completion_support.js:4:12)
>       at cleanEnvironment (out/ui-test/xml_completion_support.js:183:16)
> ```

Which OS are you using? I tested only on Linux platforms. It looks like tests lost focus on vscode window.

apupier commented:
> Which OS are you using? I tested only on Linux platforms. It looks like tests lost focus on vscode window.

Fedora 30
the first failure is different and I think that after the first failure, VS Code window is closed. (not completely sure about the timing but this is my impression when looking at it)

djelinek commented:
and opened camel-context.xml from resources really contains all lines (more then one)? I am wondering why wouldn't return all lines :(

apupier commented:
> and opened camel-context.xml from resources really contains all lines (more then one)?

yes

> I am wondering why wouldn't return all lines :(

not idea either :(
My first thought was maybe due to end of line that are different but as you tested on Linux it shouldn't be the case.

djelinek commented:
last idea, which version of Java are you using? I tested with **openjdk version "1.8.0_222"**. We will setup image for Fedora30 and we will try to figure it out where is problem. Thanks very much for feedback.

djelinek commented:
Hi @apupier could you please try to run UI Tests locally again? I updated vscode-extension-tester dependency to latest 2.0.0 version, which could solve your previous problems

apupier commented:
I have this error in the tests (and no test are passing)
![Screenshot from 2019-09-25 15-20-43](https://user-images.githubusercontent.com/1105127/65604820-5ed49280-dfa8-11e9-92c3-ee1282a09a5d.png)

djelinek commented:
> I have this error in the tests (and no test are passing)
> ![Screenshot from 2019-09-25 15-20-43](https://user-images.githubusercontent.com/1105127/65604820-5ed49280-dfa8-11e9-92c3-ee1282a09a5d.png)

@jrichter any idea how could this happened?

jrichter1 commented:
@djelinek @apupier keyboard layout issues maybe? Are you by any chance using something else than US layout?

apupier commented:
keyboard issue is a good track
I was using fr layout, I switched to English (US) and it is changing the path:
![Screenshot from 2019-09-25 15-39-40](https://user-images.githubusercontent.com/1105127/65606322-f4712180-dfaa-11e9-9ff2-0c444513a34e.png)


I have an Azerty keyboard, maybe come from that.

jrichter1 commented:
If only there was a keypress library that worked with different layouts * sigh *

djelinek commented:
@apupier could you please try to modify sleep in tests to higher value? and run it again with new clipboard tester modifications? 

```
it('Auto-completion for referenced ID of "direct" component', async function () {
  this.timeout(20000);
  const editor = new TextEditor(new EditorView(), CAMEL_ROUTE_XML);

  await editor.typeText(6, 29, DIRECT_COMPONENT_NAME);
  const contentAssist = await editor.toggleContentAssist(true) as ContentAssist;
**await new DefaultWait().sleep(150);**
  assert.isTrue(await contentAssist.hasItem(DIRECT_COMPONENT_NAME));
  const direct = await contentAssist.getItem(DIRECT_COMPONENT_NAME);
  await direct.click();
  assert.equal('<to id="_toID" uri="direct:testName"/>', (await editor.getTextAtLine(6)).trim());
});
```

djelinek commented:
I will need to replace this with waiting condition  anyway

apupier commented:
> could you please try to modify sleep in tests to higher value? and run it again with new clipboard tester modifications?

effectively, passing with an higher value

> I will need to replace this with waiting condition anyway

agree

djelinek commented:
cool, I will rework that, thanks for feedback

djelinek commented:
Hi @apupier I already did some changes

1. replaced static sleeps with waiting conditions for Content Assist
2. added new tests for Extensions view info about installed extension (only staging version info at this moment)
3. regarding (2.) I had to modify launching of unit tests but everything works identical

apupier commented on package.json:
```diff
@@ -138,10 +140,14 @@
     "vsce": "^1.66.0",
-    "vscode": "^1.1.36"
+    "vscode": "^1.1.36",
+    "vscode-extension-tester": "2.0.0",
```
why not using the `^` like other version declaration?

djelinek replied:
solved, uses same declaration

apupier commented on package.json:
```diff
@@ -138,10 +140,14 @@
+    "vscode": "^1.1.36",
+    "vscode-extension-tester": "2.0.0",
+    "vscode-test": "^1.2.0",
+    "vscode-uitests-tooling": "1.0.5"
```
why not using the `^` like other version declaration?

djelinek replied:
solved, uses same declaration

djelinek commented on tsconfig.json:
```diff
@@ -7,13 +7,16 @@
 		"outDir": "out",
-		"sourceMap": true
+		"sourceMap": true,
+		"resolveJsonModule": true,
```
**"sourceMap"** - not needed, I tested some stuff around and then I forget to remove that

**"resolveJsonModule"** - is needed for possibility of UI testing properties from 'package.json' file directly according to latest version ('package.json' file vs displayed extension info). But this property also changing hierarchy of 'out/' folder. I don't know completely why, but my humble tip is because of raw project hierarchy where 'package.json' has different location then extension.ts and others... 

**"esModuleInterop": true** - was needed as consequence of resolveJsonModule compile errors but I found solution in different syntax 

from
```
import pjson from ....
```
to
```
import * as pjson from .....
```

djelinek commented:
also updated vscode-extension-tester to latest 2.0.2 version which contains bug fixes

djelinek commented:
Thank you very much for your valuable feedback.
